hooks/react-client/useNavigationState.ts

```diff
@@ -0,0 +1,40 @@
+'use client';
+
+import type { RefObject } from 'react';
+import { useContext, useEffect } from 'react';
+
+import { NavigationStateContext } from '@/providers/navigationStateProvider';
+import { debounce } from '@/util/debounce';
+
+const useNavigationState = <T extends HTMLElement>(
+  id: string,
+  ref: RefObject<T>,
+  debounceTime = 300
+) => {
+  const navigationState = useContext(NavigationStateContext);
+
+  const handleScroll = debounce(() => {
+    if (ref.current) {
+      navigationState[id] = {
+        x: ref.current.scrollLeft,
+        y: ref.current.scrollTop,
+      };
+    }
+  }, debounceTime);
+
+  useEffect(() => {
+    const element = ref.current;
```

Comment:
No need to create a variable for the "element".

Can you also wrap all the logic behind a `if (ref.current)`?

So that the rest of the logic is behind that, and you don't need to use `?` which pollute the bundle

Reply:
Without using a variable for the element, eslint says that the ref.current value in the cleanup function is likely not the same as in the effect. (https://github.com/facebook/react/issues/14920#issuecomment-471834909)

I can wrap this in `if (element)`, I think it's better than overriding another exhaustive-dep warning and having a potential bug.

Reply:
I see, that makes sense.